packages/xo-web/src/xo-app/settings/logs/index.js

```diff
@@ -26,6 +26,24 @@ const formatMessage = data =>
     2
   )}\n${JSON.stringify(data.error, null, 2).replace(/\\n/g, '\n')}\n\`\`\``
 
+const formatedLog = log =>
+  `${log.data.method}\n${JSON.stringify(
+    log.data.params,
+    null,
+    2
+  )}\n${JSON.stringify(log.data.error, null, 2).replace(/\\n/g, '\n')}`
+
+const downloadLog = log => {
+  const element = document.createElement('a')
+  // eslint-disable-next-line no-undef
+  const file = new Blob([formatedLog(log)], {
+    type: 'text/plain',
+  })
+  element.href = URL.createObjectURL(file)
+  element.download = `${new Date(log.time).toISOString()} - XO.log`
+  element.click()
```

Comment:
It works without having to add the anchor to `body`?

Reply:
It works on Chrome, but I just tested on Firefox after your comment, and no :/

Reply:
```js
element.style.display = 'none'
document.body.appendChild(element)
element.click()
document.body.removeChild(a)
```